beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -24,20 +116,217 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpb.BlockHeadersR
 func (bs *Server) SubmitBlock(ctx context.Context, req *ethpb.BeaconBlockContainer) (*ptypes.Empty, error) {
-	return nil, errors.New("unimplemented")
+	blk := req.Message
+	root, err := blk.HashTreeRoot()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not tree hash block: %v", err)
+	}
+
+	v1alpha1Block, err := migration.V1ToV1Alpha1Block(&ethpb.SignedBeaconBlock{Block: blk, Signature: req.Signature})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not convert block to v1")
+	}
+
+	// Do not block proposal critical path with debug logging or block feed updates.
+	defer func() {
+		log.WithField("blockRoot", fmt.Sprintf("%#x", bytesutil.Trunc(root[:]))).Debugf(
+			"Block proposal received via RPC")
+		bs.BlockNotifier.BlockFeed().Send(&feed.Event{
+			Type: blockfeed.ReceivedBlock,
+			Data: &blockfeed.ReceivedBlockData{SignedBlock: v1alpha1Block},
+		})
+	}()
+
+	// Broadcast the new block to the network.
+	if err := bs.Broadcaster.Broadcast(ctx, v1alpha1Block); err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not broadcast block: %v", err)
+	}
+
+	if err := bs.BlockReceiver.ReceiveBlock(ctx, v1alpha1Block, root); err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not process beacon block: %v", err)
+	}
+
+	return &ptypes.Empty{}, nil
 }
 
 // GetBlock retrieves block details for given block id.
 func (bs *Server) GetBlock(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.BlockResponse, error) {
-	return nil, errors.New("unimplemented")
+	blk, err := bs.blockFromBlockID(ctx, req.BlockId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get block from block ID: %v", err)
+	}
+	if blk == nil {
+		return nil, status.Errorf(codes.NotFound, "Could not find requested block")
+	}
+
+	v1Block, err := migration.V1Alpha1ToV1Block(blk)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not convert block to v1")
+	}
+
+	return &ethpb.BlockResponse{
+		Data: &ethpb.BeaconBlockContainer{
+			Message:   v1Block.Block,
+			Signature: blk.Signature,
+		},
+	}, nil
 }
 
 // GetBlockRoot retrieves hashTreeRoot of BeaconBlock/BeaconBlockHeader.
 func (bs *Server) GetBlockRoot(ctx context.Context, req *ethpb.BlockRequest) (*ethpb.BlockRootResponse, error) {
-	return nil, errors.New("unimplemented")
+	var root []byte
+	var err error
+	switch string(req.BlockId) {
+	case "head":
+		root, err = bs.HeadFetcher.HeadRoot(ctx)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not retrieve head block: %v", err)
+		}
+		if root == nil {
+			return nil, status.Errorf(codes.NotFound, "No head root was found")
+		}
+	case "finalized":
+		finalized := bs.FinalizationFetcher.FinalizedCheckpt()
+		if finalized == nil {
+			return nil, status.Errorf(codes.NotFound, "No finalized root was found")
+		}
+		root = finalized.Root
+	case "genesis":
+		blk, err := bs.BeaconDB.GenesisBlock(ctx)
```

Comment:
Use a ChainInfoFetcher and you can get info about head, genesis, finalized checkpoint, and canonical. Everything you need basically. 

https://github.com/prysmaticlabs/prysm/blob/c0ed43d920c3a39bb89df453596b1d3e34c067f7/beacon-chain/blockchain/chain_info.go#L20-L25

Actually, genesisFetcher does not provide the genesis block, but it ought to do so IMO.